checker/src/main/java/org/checkerframework/checker/nullness/collection-object-parameters-may-be-null.astub

```diff
@@ -0,0 +1,254 @@
+// Lists:
+//   ArrayList
+//   LinkedList
+//   Vector
+//     Stack
+// Maps:
+//   HashMap
+//     LinkedHashMap
+//     PrinterStateReasons
+//   IdentityHashMap
+//   WeakHashMap
+// Sets:
+//   HashSet
+//     JobStateReasons
+//     LinkedHashSet
+//
+// Special cases:
+//   EnumMap:
+//     "Attempts to insert a null key will throw NullPointerException. Attempts to test for the
+//     presence of a null key or to remove one will, however, function properly. Null values are
+//     permitted."
+//
+// Here are some exceptions: methods whose specification is ambiguous (says "such that o.equals(e)"
+// or "such that Objects.equals(o, e)") but can be interpreted to say that null is a permitted
+// value:
+//
+//   ArrayBlockingQueue
+//   ArrayDeque
+//   ConcurrentLinkedDeque
+//   ConcurrentLinkedQueue
+//   CopyOnWriteArrayList
+//   CopyOnWriteArraySet
+//   LinkedBlockingDeque
+//   LinkedBlockingQueue
+//   LinkedTransferQueue
+//   PriorityBlockingQueue
+//   PriorityQueue
+//     contains
+//     remove
+//
+//   ArrayDeque
+//     removeFirstOccurrence
+//     removeLastOccurrence
+//
+//   Collections
+//     frequency
+//
+// Special cases:
+//   EnumMap:
+//     "Attempts to insert a null key will throw NullPointerException. Attempts to test for the
+//     presence of a null key or to remove one will, however, function properly. Null values are
+//     permitted."
+//
+// Many additional method implementations are null-safe, but the specification does not guarantee
+// that all overriding implementations in subclasses will be.
+
+
+package java.util;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+class AbstractCollection {
+    @Pure public boolean contains(@Nullable Object o);
+    public boolean remove(@Nullable Object o);
+    @Pure public boolean containsAll(Collection<?> c);
+    public boolean removeAll(Collection<?> c);
+    public boolean retainAll(Collection<?> c);
+}
+
+public abstract class AbstractList<E> extends AbstractCollection<E> implements List<E> {
+    public int indexOf(@Nullable Object o);
+    public int lastIndexOf(@Nullable Object o);
+}
+
+class AbstractMap {
+    public boolean containsValue(@Nullable Object value);
+    public boolean containsKey(@Nullable Object key);
+    public @Nullable V get(@Nullable Object key);
+    public @Nullable V remove(@Nullable Object key);
+}
+
+class AbstractSet {
+  public boolean removeAll(Collection<?> c);
+}
+
+public class ArrayList<E> extends AbstractList<E>
+        implements List<E>, RandomAccess, Cloneable, java.io.Serializable
+{
+    public boolean contains(@Nullable Object o);
+    public int indexOf(@Nullable Object o);
+    public int lastIndexOf(@Nullable Object o);
+    public boolean remove(@Nullable Object o);
+}
+
+public interface Collection<E> extends Iterable<E> {
+    boolean contains(@Nullable Object o);
+    boolean remove(@Nullable Object o);
+    boolean containsAll(Collection<? extends @Nullable Object> c);
```

Comment:
I notice a mix of `Collection<?>` and `Collection<? extends @Nullable Object>` in this stub file. I don't know if that makes any difference to the tool (or to humans).